compiler/plugins/target/AMD-AIE/aie/CMakeLists.txt

```diff
@@ -0,0 +1,128 @@
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+set(IREE_PACKAGE_ROOT_DIR "${CMAKE_CURRENT_LIST_DIR}")
+set(IREE_PACKAGE_ROOT_PREFIX "iree::target::amd-aie::aie")
+
+list(APPEND IREE_COMPILER_TABLEGEN_INCLUDE_DIRS
+    "${IREE_MLIR_AIE_SOURCE_DIR}/include")
+include_directories("${IREE_MLIR_AIE_SOURCE_DIR}/include")
+
+iree_cc_library(
+  NAME
+    defs
+  INCLUDES
+    "${IREE_MLIR_AIE_SOURCE_DIR}/include"
+)
+
+###############################################################################
+# AIE Dialect
+###############################################################################
+
+iree_cc_library(
+  NAME
+    AIEDialectIR
+  SRCS
+    ${IREE_MLIR_AIE_SOURCE_DIR}/lib/Dialect/AIE/IR/AIEDialect.cpp
+    ${IREE_MLIR_AIE_SOURCE_DIR}/lib/Dialect/AIE/IR/AIETargetModel.cpp
+  DEPS
+    ::defs
+    ::AIEAttrsGen
+    ::AIEDialectGen
+    ::AIEInterfacesGen
+    ::AIEOpsGen
+    ::AIETypesGen
+    ::AIETypesGen2
+    MLIRIR
+)
+
+iree_tablegen_library(
+  NAME
+    AIEAttrsGen
+  TD_FILE
+    "${IREE_MLIR_AIE_SOURCE_DIR}/include/aie/Dialect/AIE/IR/AIEAttrs.td"
+  OUTS
+    -gen-attrdef-decls Dialect/AIE/IR/AIEAttrs.h.inc
+    -gen-attrdef-defs Dialect/AIE/IR/AIEAttrs.cpp.inc
+    -gen-enum-decls Dialect/AIE/IR/AIEEnums.h.inc
+    -gen-enum-defs Dialect/AIE/IR/AIEEnums.cpp.inc
+)
+
+iree_tablegen_library(
+  NAME
+    AIEDialectGen
+  TD_FILE
+    "${IREE_MLIR_AIE_SOURCE_DIR}/include/aie/Dialect/AIE/IR/AIE.td"
+  OUTS
+    -gen-dialect-decls Dialect/AIE/IR/AIEDialect.h.inc
+    -gen-dialect-defs Dialect/AIE/IR/AIEDialect.cpp.inc
+)
+
+iree_tablegen_library(
+  NAME
+    AIEInterfacesGen
+  TD_FILE
+    "${IREE_MLIR_AIE_SOURCE_DIR}/include/aie/Dialect/AIE/IR/AIEInterfaces.td"
+  OUTS
+    -gen-op-interface-decls Dialect/AIE/IR/AIEInterfaces.h.inc
+    -gen-op-interface-defs Dialect/AIE/IR/AIEInterfaces.cpp.inc
+)
+
+iree_tablegen_library(
+  NAME
+    AIEOpsGen
+  TD_FILE
+    "${IREE_MLIR_AIE_SOURCE_DIR}/include/aie/Dialect/AIE/IR/AIEOps.td"
+  OUTS
+    -gen-op-decls Dialect/AIE/IR/AIEOps.h.inc
+    -gen-op-defs Dialect/AIE/IR/AIEOps.cpp.inc
+)
+
+iree_tablegen_library(
+  NAME
+    AIETypesGen
+  TD_FILE
+    "${IREE_MLIR_AIE_SOURCE_DIR}/include/aie/Dialect/AIE/IR/AIETypes.td"
+  OUTS
+    -gen-typedef-decls -typedefs-dialect=AIE Dialect/AIE/IR/AIETypes.h.inc
+    -gen-typedef-defs -typedefs-dialect=AIE Dialect/AIE/IR/AIETypes.cpp.inc
+)
+
+## AIE generates the same type file with multiple names.
+iree_tablegen_library(
+  NAME
+    AIETypesGen2
+  TD_FILE
+    "${IREE_MLIR_AIE_SOURCE_DIR}/include/aie/Dialect/AIE/IR/AIETypes.td"
+  OUTS
+    -gen-typedef-decls -typedefs-dialect=AIE Dialect/AIE/IR/AIETypesGen.h.inc
+    -gen-typedef-defs -typedefs-dialect=AIE Dialect/AIE/IR/AIETypesGen.cpp.inc
+)
+
```

Comment:
This seems bogus...  Are we really doing that?

Reply:
Yes AFAICS https://github.com/Xilinx/mlir-aie/blob/main/include/aie/Dialect/AIE/IR/CMakeLists.txt 

The add_mlir_dialect generates `AIRTypes.h.inc` and there is a special rule for `AIETypesGen.h.inc`

Reply:
To me this looks like a merge conflict that didn't get resolved properly.